engineio/transport/factory.go

```diff
@@ -0,0 +1,32 @@
+package transport
+
+import (
+	"fmt"
+	"time"
+)
+
+var creators = make(map[Name]Creator)
+
+// Name is a name of transport.
+type Name string
+
+const (
```

Comment:
in this case we can use go gen with stringer. loop up for [this](https://blog.golang.org/generate)

Reply:
Well, I don't want to use `int` here because names don't have an integer meaning. [The usage](https://github.com/socketio/engine.io-protocol#sample-session) is only string. I hope to meet this usage with type.

Reply:
I understood your motivation, it look like good